presqt/osf/classes/storage_folder.py

```diff
@@ -59,6 +75,188 @@ def get_all_files(self):
+            return self.get_folder_by_name(folder_name)
+
+        elif response.status_code == 201:
+            return self.get_folder_by_name(folder_name)
+
+        else:
+            raise PresQTResponseException(
+                "Response has status code {} while creating folder {}".format(response.status_code,
+                                                                              folder_name),
+                status.HTTP_400_BAD_REQUEST)
+
+    def create_file(self, file_name, file_to_write, file_duplicate_action):
+        """
+        Upload a file to a container.
+
+        Parameters
+        ----------
+        file_name : str
+            Name of the file to create.
+        file_to_write : binary file
+            File to create.
+        file_duplicate_action : str
+            Flag for how to handle the case of the file already existing.
+
+        Returns
+        -------
+        Class instance of the created file.
+        """
+        # When uploading a large file (>a few MB) that already exists
+        # we sometimes get a ConnectionError instead of a status == 409.
+        connection_error = False
+        try:
+            response = self.put(self._new_file_url, params={'name': file_name}, data=file_to_write)
+        except ConnectionError:
+            connection_error = True
+
+        # If the file is a duplicate then either ignore or update it
+        if connection_error or response.status_code == 409:
+            original_file = self.get_file_by_name(file_name)
+
+            if file_duplicate_action == 'ignore':
+                return 'ignored', original_file
+
+            elif file_duplicate_action == 'update':
+                # Only attempt to update the file if the new file is different than the original
+                if hash_generator(file_to_write, 'md5') != original_file.hashes['md5']:
+                    response = self.get_file_by_name(file_name).update(file_to_write)
+
+                    if response.status_code == 200:
+                        return 'updated', self.get_file_by_name(file_name)
+                    else:
+                        raise PresQTResponseException(
+                            "Response has status code {} while updating file {}".format(
+                                response.status_code, file_name), status.HTTP_400_BAD_REQUEST)
+                else:
+                    return 'ignored', original_file
+
+        # File uploaded successfully
+        elif response.status_code == 201:
+            return 'created', self.get_file_by_name(file_name)
+
+        else:
+            raise PresQTResponseException(
+                "Response has status code {} while creating file {}".format(response.status_code,
+                                                                              file_name),
+                status.HTTP_400_BAD_REQUEST)
+
+
+    def create_directory(self, directory_path, file_duplicate_action, file_hashes,
```

Comment:
The name threw me off a bit here. It seems to me that this creates multiple directories/folders where the name of the method made me think it was just one.